examples/algorithm_progression.py

```diff
@@ -0,0 +1,179 @@
+        '-w',
+        '--num_workers',
+        required=False,
+        default=4,
+        type=int,
+        help='Number of workers to use for processing alogrithm iterations',
+    )
+    argparser.add_argument(
+        '-p',
+        '--param',
+        action='append',
+        required=False,
+        nargs='*',
+        help='A parameter to pass to the algorithm; '
+             'instead of using the default value, '
+             'Pass every item in the number space, '
+             'specified as `--param=param_name,start,end,num_items[,open]`',
+    )
+    return argparser
+
+
+def apply_algorithm(algorithm, input_filename, output_dir, params, param_space):
+    iteration_id = [
+        list(values).index(params[i]) for i, values in enumerate(param_space.values())
+        if len(values) > 1
+    ]
+    filename = f'{algorithm.__name__}_{"_".join([str(v) for v in iteration_id])}.tiff'
+    filepath = Path(output_dir, filename)
+
+    desc = dict(
+        **{'path': filename},
+        **{VARIABLE_LAYERS[i]: v for i, v in enumerate(iteration_id)}
+    )
+
+    source = large_image.open(input_filename)
+    new_source = large_image.new()
+    for tile in source.tileIterator(
+        format=large_image.tilesource.TILE_FORMAT_NUMPY,
+        tile_size=dict(width=2048, height=2048),
+    ):
+        altered_data = algorithm(tile['tile'], *params)
+    new_source.addTile(altered_data, tile['x'], tile['y'])
+    new_source.write(str(filepath), lossy=False)
+    return desc
+
+
+def sweep_algorithm(algorithm, input_filename, input_params, output_dir, max_workers):
+    algorithm_name = algorithm.__name__.replace('_', ' ').title()
+    yaml_dict = {
+        'name': f'{algorithm_name} iterative results',
+        'description': f'{algorithm_name} algorithm performed on {input_filename}',
+        'sources': [{'path': input_filename}],
+    }
+    param_desc = [
+        f'{VARIABLE_LAYERS[i]} represents change in {n} as an index of {p}'
+        if len(p) > 1 and i < len(VARIABLE_LAYERS)
+        else f'{n} remains unchanged per run with a value of {p[0]}'
+        for i, (n, p) in enumerate(input_params.items())
+    ]
+    if len(param_desc) > 0:
+        yaml_dict['description'] += f', where {",".join(param_desc)}'
+
+    param_space_combos = list(itertools.product(*input_params.values()))
+    print(f'Beginning {len(param_space_combos)} runs on {max_workers} workers...')
+    with concurrent.futures.ProcessPoolExecutor(max_workers=max_workers) as executor:
+        # cannot pass source through submitted task; it is unpickleable
+        futures = [
+            executor.submit(
+                apply_algorithm,
+                algorithm,
+                input_filename,
+                output_dir,
+                combo,
+                input_params
+            )
+            for combo in param_space_combos
+        ]
+        for future in concurrent.futures.as_completed(futures):
+            print('Completed run.')
```

Comment:
Can we print the actual number of frames processed?  Maybe maintain a done counter and `print(f'Completed {done}/{len(param_space_combos)}')`

Reply:
Changed in [29902be](https://github.com/girder/large_image/pull/1051/commits/29902be9031c53f83d89011752c5e1c0481e5071)